File: app/src/main/java/com/example/weizhenbin/wangebug/views/emptyview/IEmptyViewProxy.java
package com.example.weizhenbin.wangebug.views.emptyview;

import android.view.View;

/**
 * Created by weizhenbin on 2018/9/27.
 */

public interface IEmptyViewProxy {
     void setEmptyText(String text);

     void setActionText(String text);

     void setAction(View.OnClickListener clickListener);
     void setEventVisibility(int visibility);

     void loading();
     void loading(boolean isShowProgressBar);

     void emptyData();

     void emptyData(boolean showAction);

     void setProgressBarVisibility(int visibility);
}

File: app/src/main/java/com/example/weizhenbin/wangebug/views/emptyview/IEmptyViewProxy.kt
package com.example.weizhenbin.wangebug.views.emptyview

import android.view.View

/**
 * Created by weizhenbin on 2018/9/27.
 */

interface IEmptyViewProxy {
    fun setEmptyText(text: String)

    fun setActionText(text: String)

    fun setAction(clickListener: View.OnClickListener?)
    fun setEventVisibility(visibility: Int)

    fun loading()
    fun loading(isShowProgressBar: Boolean)

    fun emptyData()

    fun emptyData(showAction: Boolean)

    fun setProgressBarVisibility(visibility: Int)
}
